diff -Nur /home/vmware/misc/ndk-sash/src/android-ndk-src-2012-06-30+445d63b/binutils/binutils-2.21/libiberty/getpagesize.c binutils-2.21/libiberty/getpagesize.c
--- s/libiberty/getpagesize.c-	2012-07-05 03:27:46.000000000 +0700
+++ s/libiberty/getpagesize.c	2012-07-05 03:48:35.536254573 +0700
@@ -60,12 +60,13 @@
 # endif /* PAGESIZE */
 #endif /* GNU_OUR_PAGESIZE */
 
+#ifndef __BIONIC__
 int
 getpagesize (void)
 {
   return (GNU_OUR_PAGESIZE);
 }
-
+#endif
 #else /* VMS */
 
 #if 0	/* older distributions of gcc-vms are missing <syidef.h> */
--- s/gas/read.c-	2013-03-04 11:58:23.302649836 +0700
+++ s/gas/read.c	2013-03-04 11:58:35.370650160 +0700
@@ -1641,11 +1641,13 @@
 	}
       *name = 0;
 
+#ifndef __BIONIC__
       /* Since quoted symbol names can contain non-ASCII characters,
 	 check the string and warn if it cannot be recognised by the
 	 current character set.  */
       if (mbstowcs (NULL, name, len) == (size_t) -1)
 	as_warn (_("symbol name not recognised in the current locale"));
+#endif
     }
   else if (is_name_beginner (c) || c == '\001')
     {
--- s/bfd/configure-	2012-07-17 21:17:10.544371029 +0700
+++ s/bfd/configure	2012-07-17 21:17:21.140423570 +0700
@@ -18016,7 +18016,7 @@
 COREFILE=
 COREFLAG=
 TRAD_HEADER=
-if test "${target}" = "${host}"; then
+if false ; then
   case "${host}" in
   alpha*-*-freebsd* | alpha*-*-kfreebsd*-gnu)
 	COREFILE=''
