diff -Nur /home/sash/Work/android/ndk-source/llvm-3.3/llvm/utils/unittest/Makefile llvm-3.3/utils/unittest/Makefile
--- /home/sash/Work/android/ndk-source/llvm-3.3/llvm/utils/unittest/Makefile	2013-10-22 15:38:25.896713569 +0700
+++ llvm-3.3/llvm/utils/unittest/Makefile	2013-10-24 02:19:21.777097229 +0700
@@ -8,6 +8,6 @@
 ##===----------------------------------------------------------------------===##
 
 LEVEL = ../..
-PARALLEL_DIRS = googletest UnitTestMain
+PARALLEL_DIRS =
 
 include $(LEVEL)/Makefile.common
diff -Nur /home/sash/Work/android/ndk-source/llvm-3.3/clang/Makefile llvm-3.3/clang/Makefile
--- /home/sash/Work/android/ndk-source/llvm-3.3/clang/Makefile	2013-10-22 15:38:12.580713017 +0700
+++ llvm-3.3/clang/Makefile	2013-10-24 13:01:58.299475757 +0700
@@ -14,7 +14,7 @@
 
 IS_TOP_LEVEL := 1
 CLANG_LEVEL := .
-DIRS := utils/TableGen include lib tools runtime docs unittests
+DIRS := utils/TableGen include lib tools runtime docs
 
 PARALLEL_DIRS :=
 
diff -Nur /home/sash/Work/android/ndk-source/llvm-3.3/clang/tools/libclang/CIndexer.cpp llvm-3.3/clang/tools/libclang/CIndexer.cpp
--- /home/sash/Work/android/ndk-source/llvm-3.3/clang/tools/libclang/CIndexer.cpp	2013-10-22 15:38:15.332713131 +0700
+++ llvm-3.3/clang/tools/libclang/CIndexer.cpp	2013-10-24 12:44:04.043520934 +0700
@@ -69,6 +69,7 @@
   llvm::sys::Path LibClangPath(path);
   LibClangPath.eraseComponent();
 #else
+#ifndef __BIONIC__
   // This silly cast below avoids a C++ warning.
   Dl_info info;
   if (dladdr((void *)(uintptr_t)clang_createTranslationUnit, &info) == 0)
@@ -78,6 +79,16 @@
   
   // We now have the CIndex directory, locate clang relative to it.
   LibClangPath.eraseComponent();
+#else
+  char *clangLib = getenv("CCTOOLS_CLANG_LIB");
+
+  if (!clangLib)
+    llvm_unreachable("Call to getenv(\"CCTOOLS_CLANG_LIB\") failed");
+
+  llvm::sys::Path LibClangPath(clangLib);
+
+  LibClangPath.eraseComponent();
+#endif
 #endif
   
   LibClangPath.appendComponent("clang");
